x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -314,7 +330,35 @@ public NamedExpression visitQualifiedNamePattern(EsqlBaseParser.QualifiedNamePat
                 objects.add(".");
             }
 
-            String patternContext = patterns.get(i).ID_PATTERN().getText();
+            String patternContext = "";
+            if (patterns.get(i).ID_PATTERN() != null) {
```

Comment:
Extract the patterns.get(i) into a local variable which gets reused throughout this block. Same for ID_PATTERN and other method calls.